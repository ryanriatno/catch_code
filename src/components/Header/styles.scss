.site-header {
  display: flex;
  justify-content: space-between;
  padding: 2em 0;
  margin: 0 auto;
  width: 100%;
  &__logo {
    background: url('../../assets/img/logo.png');
    background-size: 112px 49px;
    background-position: center;
    width: 112px;
    height: 50px;
  }
  &__sort {
    background: transparent;
    border: none;
    &:focus,
    &:active {
      outline: none;
    }
  }
}