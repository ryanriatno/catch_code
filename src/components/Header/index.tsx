import React from 'react'
import './styles.scss'

interface Props {
  handleSortingPrice: (e) => void
}
const Header: React.FC<Props> = (props) => {
  return (
    <header className="site-header">
      <div className="site-header__logo" />
      <select className="site-header__sort" name="sort" id="sort" onChange={(e) => props.handleSortingPrice(e)}>
        <option value="asc">Highest Price</option>
        <option value="desc">Lowest Price</option>
      </select>
    </header>
  )
}

export default Header