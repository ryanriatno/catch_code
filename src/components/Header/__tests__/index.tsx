import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
// @ts-ignore
import Header from '../index'

describe("<Header /> test case", () => {
  test("Render Header without error", () => {
    const { container } = render(<Header handleSortingPrice={() => { return }} />);
    expect(container).toMatchSnapshot();
  });
});