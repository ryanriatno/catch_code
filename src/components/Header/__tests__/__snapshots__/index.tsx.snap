// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> test case Render Header without error 1`] = `
<div>
  <header
    class="site-header"
  >
    <div
      class="site-header__logo"
    />
    <select
      class="site-header__sort"
      id="sort"
      name="sort"
    >
      <option
        value="asc"
      >
        Highest Price
      </option>
      <option
        value="desc"
      >
        Lowest Price
      </option>
    </select>
  </header>
</div>
`;
